src/hera/workflows/script.py

```diff
@@ -156,31 +157,66 @@ def _build_inputs(self) -> Optional[ModelInputs]:
             else:
                 func_parameters = _get_parameters_from_callable(self.source)
 
-        if inputs is None and not func_parameters and not func_artifacts:
+        return self._process_io(inputs, func_parameters, func_artifacts, False)
+
+    def _build_outputs(self) -> Optional[ModelOutputs]:
```

Comment:
Same comment on this as [here](https://github.com/argoproj-labs/hera/pull/741/files#r1287511036)